### DISCUSSION TITLE
Conditional row formatting

### DISCUSSION THREAD

**Discussion by AnnePdM** [JanMarvin/openxlsx2#433] [Q&A] [answered]
I would like to be able to format the text (say red) in a whole excel row based on a value in one (or more) of the columns. Can anyone help? To do this, I was playing around with the `mtcars` dataset. For example, I would like to highlight the rows where `mpg` is less than 20 and `vs` is equal to one. I think that is three rows. 
The only examples I can find seem to relate to highlighting single cells rather than whole rows.

**Comment by JanMarvin** [comment, marked as answer]
Hi @AnnePdM , something like this?

``` r
library(openxlsx2)

# create style
negStyle <- create_dxfs_style(font_color = wb_colour(hex = "FF9C0006"),
                              bgFill = wb_colour(hex = "FFFFC7CE"))

# create custom row
dat <- mtcars
dat$hl <- dat$mpg < 20 & dat$vs == 1

# get excel column name
rulerow <- int2col(ncol(dat))

# create workbook add data
wb <- wb_workbook()$
  add_worksheet()$
  add_data(x = dat)

# add style to workbook
wb$styles_mgr$add(negStyle, "negStyle")

# add conditional formatting
wb$add_conditional_formatting(
  "Sheet 1",
  cols = seq_along(dat),
  rows = seq_len(nrow(dat)) + 1,
  rule = sprintf("$%s2=TRUE", rulerow),
  style = "negStyle"
)
```

**Comment by AnnePdM** [comment]
Thanks that's perfect, I would never have got there on my own!

**Reply by JanMarvin** [reply]
Glad it helps. It's a learning experience for all of us :)

**Comment by JanMarvin** [comment]
Another solution is this - stumbled over [here](https://stackoverflow.com/questions/74726485/openxlsx-conditional-formatting-with-multiple-rules/74726809#comment131886802_74726485) (tested with LibreOffice). Using `AND()` to combine two Excel formulas. Wasn't even aware that this possibility exists.

```R
library(openxlsx2)

# create style
negStyle <- create_dxfs_style(font_color = wb_colour(hex = "FF9C0006"),
                              bgFill = wb_colour(hex = "FFFFC7CE"))

# create workbook add data
wb <- wb_workbook()$
  add_worksheet()$
  add_data(x = mtcars)

# add style to workbook
wb$styles_mgr$add(negStyle, "negStyle")

# add conditional formatting
wb$add_conditional_formatting(
  "Sheet 1",
  cols = seq_along(mtcars),
  rows = seq_len(nrow(mtcars)) + 1,
  rule = "AND($A2<20,$H2=1)",
  style = "negStyle"
)
```

**Comment by AnnePdM** [comment]
Thanks!
